lib/charms/hydra/v0/hydra_endpoints_info.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Interface library for sharing hydra endpoints.
+
+This library provides a Python API for both requesting and providing public and admin endpoints.
+
+## Getting Started
+To get started using the library, you need to fetch the library using `charmcraft`.
+```shell
+cd some-charm
+charmcraft fetch-lib charms.hydra.v0.hydra_endpoints_info
+```
+
+To use the library from the provider side (Hydra):
+In the `metadata.yaml` of the charm, add the following:
+```yaml
+provides:
+  endpoint-info:
+    interface: hydra-endpoints-info
+    description: Provides API endpoints to a related application
+```
+
+To use the library from the requirer side (Kratos):
+In the `metadata.yaml` of the charm, add the following:
+```yaml
+requires:
+  endpoint-info:
+    interface: hydra-endpoints-info
+    limit: 1
+```
+Then, to initialise the library:
+```python
+from charms.hydra.v0.hydra_endpoints_info import (
+    HydraEndpointsRelationError,
+    HydraEndpointsRequirer,
+)
+
+Class SomeCharm(CharmBase):
+    def __init__(self, *args):
+        self.hydra_endpoints_relation = HydraEndpointsRequirer(self)
+        self.framework.observe(self.on.some_event_emitted, self.some_event_function)
+    def some_event_function():
+        # fetch the relation info
+        try:
+            hydra_data = self.hydra_endpoints_relation.get_relation_data()
+        except HydraEndpointsRelationError as error:
+            ...
+```
+
+"""
+
+import logging
+
+from ops.framework import Object
+from ops.model import Application
+
+# TODO: Update once the lib is published
+# # The unique Charmhub library identifier, never change it
+# LIBID = ""
+#
+# # Increment this major API version when introducing breaking changes
+# LIBAPI = 0
+#
+# # Increment this PATCH version before using `charmcraft publish-lib` or reset
```

Comment:
Those will be auto-generated when you create the lib `charmcraft create-lib`, not the `publish-lib` meaning that it should not be done in a different PR.